src/OrchardCore.Modules/OrchardCore.OpenId/Drivers/OpenIdValidationSettingsDisplayDriver.cs

```diff
@@ -71,7 +73,7 @@ public class OpenIdValidationSettingsDisplayDriver : SectionDisplayDriver<ISite,
                 model.Audience = settings.Audience;
                 model.Tenant = settings.Tenant;
 
-                model.AvailableTenants = (from tenant in _shellSettingsManager.LoadSettings().AsParallel()
+                model.AvailableTenants = (from tenant in _shellSettingsManager.LoadSettings().Where(s => s.State == TenantState.Running).AsParallel()
                                           let provider = _shellHost.GetOrCreateShellContext(tenant).ServiceProvider
```

Comment:
We should probably remove this LINQ query and replace it by a `foreach` to be able to use the `EnterServiceScope` API in a `using` block.

Reply:
Okay cool, i will do it after reviewing the other points.